packages/sn-universal-header/src/components/hamburger-menu.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  ClickAwayListener,
+  createStyles,
+  IconButton,
+  Link,
+  makeStyles,
+  MenuItem,
+  MenuList,
+  Paper,
+  Theme,
+} from '@material-ui/core'
+import { Menu as MenuIcon, MenuOpen as MenuOpenIcon } from '@material-ui/icons'
+import React, { useEffect, useRef, useState } from 'react'
+import { HEADER_HEIGHT } from '.'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    iconButton: {
+      padding: 0,
+      '&:hover': {
+        background: 'none',
+      },
+    },
+    menuIcon: {
+      color: theme.palette.common.white,
+      marginRight: '20px',
+      '&:hover': {
+        color: '#a49ec5',
+      },
+    },
+    menuIconActive: {
+      marginRight: '20px',
+      color: '#a49ec5',
+    },
+    popperWrapper: {
+      position: 'absolute',
+      top: HEADER_HEIGHT,
+      left: 0,
+      height: 'fit-content',
+      width: '216px',
+    },
+    popper: {
+      backgroundColor: '#F6F6F6',
+      border: '1px solid #E2E2E2',
+    },
+    menuItem: {
+      textDecoration: 'underline',
+      color: theme.palette.primary.main,
+      fontSize: '14px',
+    },
+  }),
+)
+
+export type HamburgerMenuClassKey = Partial<ReturnType<typeof useStyles>>
+
+interface HamburgerMenuProps {
+  classes?: HamburgerMenuClassKey
+}
+
+export const HamburgerMenu: React.FunctionComponent<HamburgerMenuProps> = (props) => {
+  const classes = useStyles(props)
+
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isPopperOpen, setIsPopperOpen] = useState(false)
+
+  const handleToggle = () => {
+    setIsPopperOpen((prevOpen) => !prevOpen)
+  }
+
+  const handleClose = (event: React.MouseEvent<EventTarget>) => {
+    if (anchorRef.current?.contains(event.target as HTMLElement)) {
+      return
+    }
+
+    setIsPopperOpen(false)
+  }
+
+  const prevOpen = useRef(isPopperOpen)
+  useEffect(() => {
+    if (prevOpen.current === true && isPopperOpen === false) {
+      anchorRef.current!.focus()
+    }
+
+    prevOpen.current = isPopperOpen
+  }, [isPopperOpen])
+
+  return (
+    <div>
+      <IconButton className={classes.iconButton} onClick={handleToggle} ref={anchorRef}>
+        {isPopperOpen ? <MenuOpenIcon className={classes.menuIconActive} /> : <MenuIcon className={classes.menuIcon} />}
```

Comment:
`<MenuOpenIcon className={isPopperOpen ? classes.menuIconActive : classes.menuIcon} />`

Reply:
I have 2 icons for the 2 states, it can stay like this. discussed